freeipa/src/main/java/com/sequenceiq/freeipa/service/stack/FreeIpaHealthDetailsService.java

```diff
@@ -0,0 +1,132 @@
+package com.sequenceiq.freeipa.service.stack;
+
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+
+import com.sequenceiq.freeipa.api.v1.freeipa.stack.model.common.DetailedStackStatus;
+import com.sequenceiq.freeipa.api.v1.freeipa.stack.model.common.Status;
+import com.sequenceiq.freeipa.api.v1.freeipa.stack.model.common.instance.InstanceGroupType;
+import com.sequenceiq.freeipa.api.v1.freeipa.stack.model.health.HealthDetailsFreeIpaResponse;
+import com.sequenceiq.freeipa.api.v1.freeipa.stack.model.health.NodeHealthDetailsFreeIpaResponse;
+import com.sequenceiq.freeipa.client.FreeIpaClient;
+import com.sequenceiq.freeipa.client.FreeIpaClientException;
+import com.sequenceiq.freeipa.client.model.RPCMessage;
+import com.sequenceiq.freeipa.client.model.RPCResponse;
+import com.sequenceiq.freeipa.entity.InstanceGroup;
+import com.sequenceiq.freeipa.entity.Stack;
+import com.sequenceiq.freeipa.service.freeipa.FreeIpaClientFactory;
+
+@Service
+public class FreeIpaHealthDetailsService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(FreeIpaHealthDetailsService.class);
+
+    private static final String EXTERNAL_COMMAND_OUTPUT = "ExternalCommandOutput";
+
+    private static final String STATUS_OK = "OK";
+
+    private static final int STATUS_GROUP = 2;
+
+    private static final String MESSAGE_UNAVAILABLE = "Message Unavailable";
+
+    private static final Pattern RESULT_PATTERN = Pattern.compile("(ecure port|: TCP) \\([0-9]*\\): (.*)");
+
+    private static final Pattern NEW_NODE_PATTERN = Pattern.compile("Check connection from master to remote replica '(.[^\']*)");
+
+    @Inject
+    private StackService stackService;
+
+    @Inject
+    private FreeIpaClientFactory freeIpaClientFactory;
+
+    public HealthDetailsFreeIpaResponse getHealthDetails(String environmentCrn, String accountId) {
+        Stack stack = stackService.getByEnvironmentCrnAndAccountIdWithLists(environmentCrn, accountId);
+        String masterCN = findMasterCN(stack);
+        Optional<RPCResponse<Boolean>> rpcResponse = Optional.empty();
+        try {
+            rpcResponse = Optional.ofNullable(checkFreeIpaHealth(stack, masterCN));
```

Comment:
1. this call's during can be more than 2sec (with freeipaClientFactory more than 5 sec). All client will timeout
2. the autosyncer do the same. I think you can parse the response there and set into the database as a `statusReason`. 

This solution can provide a different status than we now in the db and produce inconsistence operation

Reply:
Per our conversation, 1. the only clients that should be using this today are the ones interacting for HA purposes so we should be able to handle the longer call times.
2. While these are close today, the status we'll need for the individual groups, and nodes in the stack is a slightly different use case.  Additionally, we might need to move away from server_conncheck, but using it for now since it's there.
There will probably be additional permutations of this structure as we get all the HA pieces lined up and once finalized, we will sync up with the describe status.  For now I think it's good to keep these separate because the users are separate, and we might need to add additional details as we work through adding HA.  I create https://jira.cloudera.com/browse/CB-5450 to track the follow up.